docs/proposals/20191017-control-plane.md

```diff
@@ -0,0 +1,495 @@
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      - [New API Types](#new-api-types)
+      - [Modifications required to existing API Types](#modifications-required-to-existing-api-types)
+      - [Behavioral Changes from v1alpha2](#behavioral-changes-from-v1alpha2)
+      - [Behaviors](#behaviors)
+        - [Create](#create)
+        - [Scale Up](#scale-up)
+        - [Scale Down](#scale-down)
+        - [Delete of the entire ControlPlane (kubectl delete controlplane my-controlplane)](#delete-of-the-entire-controlplane-kubectl-delete-controlplane-my-controlplane)
+        - [Cluster upgrade (using create-swap-and-delete)](#cluster-upgrade-using-create-swap-and-delete)
+        - [Control plane healthcheck](#control-plane-healthcheck)
+        - [Adoption of pre-v1alpha3 Control Plane Machines](#adoption-of-pre-v1alpha3-control-plane-machines)
+      - [Code organization](#code-organization)
+    - [Risks and Mitigations](#risks-and-mitigations)
+      - [etcd membership](#etcd-membership)
+  - [Design Details](#design-details)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+      - [Alpha -> Beta Graduation](#alpha---beta-graduation)
+    - [Upgrade Strategy](#upgrade-strategy)
+  - [Alternatives](#alternatives)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+The lexicon used in this document is described in more detail [here](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/reference/glossary.md).  Any discrepancies should be rectified in the main Cluster API glossary.
+Stacked Control Plane / Stacked etcd: Control plane node where etcd is colocated with the Kubernetes API server, and is running as a static pod.
+
+### References
+
+[Kubernetes Control Plane Management Overview](https://docs.google.com/document/d/1nlWKEr9OP3IeZO5W2cMd3A55ZXLOXcnc6dFu0K9_bms/edit#)
+
+## Summary
+
+Management of the control plane is currently handled in Cluster API at the infrastructure provider level (although many providers have converged on using the same implementation), based on the special annotation of standard Cluster API machines which then changes the machine bootstrapping logic within a particular infrastructure provider.
+The purpose of this document is to instead treat machine-based control plane management as a first-class component that is part of the API and Cluster, where multiple Cluster API machines will belong to a single control plane object, with its own infrastructure independent provider. The control plane covered by this document is defined as consisting of the Kubernetes API server, scheduler, controller manager, DNS and proxy services, and the underlying etcd datastore.
+
+## Motivation
+
+### Goals
+
+- To establish new resource types for control plane management
+- To support single node and multiple control plane instances
+- To enable declarative orchestrated control plane upgrades
```

Comment:
Do we need to call out that we want to support replacing control plane machines without performing Kubernetes upgrades (because we need to roll out an OS update)?